q2_qemistree/_classyfire.py

```diff
@@ -78,14 +78,16 @@ def get_classyfire_taxonomy(feature_data: pd.DataFrame) -> pd.DataFrame:
             response = requests.get(to_classyfire+str(inchikey)+'.json')
             if response.status_code == 200:
                 response = response.json()
-                classyfire_levels_sub = [level for level in classyfire_levels
-                                         if level in response.keys()]
-                if len(classyfire_levels_sub) == 0:
+                sublevels = [level for level in classyfire_levels
+                             if level in response.keys()]
+                if len(sublevels) == 0:
+                    classyfire[idx] = 'unclassified'
                     continue
                 taxonomy = [response[level]['name']
-                            if bool(response) and response[level] is not None
+                            if (level in sublevels and bool(response) and
```

Comment:
Can you expand this if/else clause assignment into the multiple line version? This seems a little bit convoluted and challenging to follow.

Reply:
Agree, one other question though, what is `bool(response)`? I *think* this is testing whether `response` is a non-empty `dict`, in which case it would be more pythonic to do `...and response and...`